File: backend/src/main/java/ca/metricalsky/winston/repository/fetch/YouTubeRequestRepository.java
package ca.metricalsky.winston.repository.fetch;

import ca.metricalsky.winston.entity.fetch.YouTubeRequestEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.OffsetDateTime;

@Repository
public interface YouTubeRequestRepository extends JpaRepository<YouTubeRequestEntity, Long> {

    int countAllByRequestedAtAfter(OffsetDateTime date);
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/repository/fetch/YouTubeRequestRepository.kt
package ca.metricalsky.winston.database.repository.fetch

import ca.metricalsky.winston.database.entity.fetch.YouTubeRequestEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.OffsetDateTime

@Repository
interface YouTubeRequestRepository : JpaRepository<YouTubeRequestEntity, Long> {

    fun countAllByRequestedAtAfter(date: OffsetDateTime): Int
}
